.github/workflows/pypi-publish.yml

```diff
@@ -0,0 +1,28 @@
+name: Build dist and publish to PyPI
+
+on: [push, pull_request]
+
+jobs:
+  pypi-publish:
+    name: Upload release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: testpypi
```

Comment:
Similarly, this should just be `pypi`.  What do you think is the best way of having a "test" action separately from this one, which should be the _main_ action?